target/linux/ath79/files/drivers/net/ethernet/atheros/ag71xx/ag71xx_main.c

```diff
@@ -1577,7 +1598,7 @@ static int ag71xx_probe(struct platform_device *pdev)
 	if (of_property_read_u32_array(np, "pll-reg", ag->pllreg, 3))
 		dev_dbg(&pdev->dev, "failed to read pll-reg property\n");
 
-	ag->pllregmap = syscon_regmap_lookup_by_phandle(np, "pll-handle");
+	ag->pllregmap = ag71xx_get_pll_regmap(np);
```

Comment:
Why is `of_clk_get` returning `-EPROBE_DEFER` here?

Reply:
Oh no, you caught me!  Okay, I admit that figuring out why the heck the switch does not work on this thing was frustrating enough that the prospect of having to dig into the kernel's clock API and driver binding logic got the better of me and I settled for a hack.  Now that you asked, though, I went down that rabbit hole and it turns out that the answer is simple - the `ref` DTS node (which is the clock associated with the PLL controller) did not have its `clock-frequency` property set, preventing the clock from being "initialized".

I dropped 69273c7559475034e0841e5389c34c34daf5da61 and added the `clock-frequency` property to the `ref` clock in e60ef9acf172198eaafffe87bd1021e3cbd69e11.  No more kernel hacks in this PR.  Thanks for keeping me honest.